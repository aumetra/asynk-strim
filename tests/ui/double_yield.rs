fn main() {
    let _stream = asynk_strim::strim_fn(|mut yielder| async move {
        let _first = yielder.yield_item(1312);
        let _second = yielder.yield_item(141);
    });
}