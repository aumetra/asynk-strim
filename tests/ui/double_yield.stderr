error[E0499]: cannot borrow `yielder` as mutable more than once at a time
 --> tests/ui/double_yield.rs:4:23
  |
3 |         let _first = yielder.yield_item(1312);
  |                      ------- first mutable borrow occurs here
4 |         let _second = yielder.yield_item(141);
  |                       ^^^^^^^ second mutable borrow occurs here
5 |     });
  |     - first borrow might be used here, when `_first` is dropped and runs the destructor for type `impl Future<Output = ()> + '_`
